src/worker.py

```diff
@@ -162,10 +163,16 @@ def __init__(self, directory, hidden=False):
 
     def start(self):
         self.emit("started", self._directory)
-        self._paths = os.listdir(self._directory)
-        self._total = len(self._paths)
-        self._index = 0
-        GLib.idle_add(self.step, priority=GLib.PRIORITY_HIGH_IDLE + 20)
+        try:
+            self._paths = os.listdir(self._directory)
+            self._total = len(self._paths)
+            self._index = 0
+            GLib.idle_add(self.step, priority=GLib.PRIORITY_HIGH_IDLE + 20)
+        except Exception as ex:
+            # This can happen if the directory doesn't exists
```

Comment:
Just use the logger for `e` (keep the `as e` naming consistent, see other except in this file), no need to display the specific error on the UI.

Reply:
Done!